src/com/gfredericks/schpec.clj

```diff
@@ -44,3 +45,11 @@
                   [name `(s/and ~spec ~@preds)]))
               key-pred-forms)]
     `(s/or ~@opts)))
+
+(defmacro alias
+  "Like clojure.core/alias, but can alias to non-existing namespaces"
+  [alias namespace-sym]
+  `(try (clojure.core/alias ~alias ~namespace-sym)
+        (catch Exception _#
+          (create-ns ~namespace-sym)
+          (clojure.core/alias ~alias ~namespace-sym))))
```

Comment:
Is there any need for this to be a macro? If not I'd rather keep it as a function to match the signature of `clojure.core/alias`.